packages/shared/src/hooks/useCompanionTrigger.tsx

```diff
@@ -0,0 +1,160 @@
+    column?: number,
+  ) => Promise<void>,
+): CompanionTrigger {
+  const isExtension = process.env.TARGET_BROWSER;
+  const { trackEvent } = useContext(AnalyticsContext);
+  const { closeModal: closeLazyModal, openModal: openLazyModal } =
+    useLazyModal();
+  const { isActionsFetched, checkHasCompleted, completeAction } = useActions();
+  const gb = useGrowthBook();
+
+  const { requestContentScripts, useContentScriptStatus } =
+    useExtensionPermission({
+      origin: 'companion modal permission button',
+    });
+
+  const { contentScriptGranted } = useContentScriptStatus();
+  const alreadyCompleted = checkHasCompleted(
+    ActionType.EngagementLoopJuly2023CompanionModal,
+  );
+
+  const handleCompleteAction = useCallback(() => {
+    if (isActionsFetched)
+      completeAction(ActionType.EngagementLoopJuly2023CompanionModal);
+  }, [isActionsFetched, completeAction]);
+
+  const closeModal = useCallback(() => {
+    closeLazyModal();
+    handleCompleteAction();
+
+    trackEvent({
+      event_name: `close companion permission popup`,
+      extra: JSON.stringify({ origin: 'automatic' }),
+    });
+  }, [closeLazyModal, handleCompleteAction, trackEvent]);
+
+  const activateCompanion = useCallback(async () => {
+    await requestContentScripts();
+    closeModal();
+    handleCompleteAction();
+  }, [requestContentScripts, closeModal, handleCompleteAction]);
+
+  const openArticle: OpenArticle = useCallback(
+    (data) => async () => {
+      await customPostClickHandler(
+        data.post,
+        data.index,
+        data.row,
+        data.column,
+      );
+
+      closeModal();
+      handleCompleteAction();
+    },
+    [customPostClickHandler, closeModal, handleCompleteAction],
+  );
+
+  const openModal = useCallback(
+    (data?: CompanionTriggerProps) => {
+      openLazyModal({
+        type: LazyModal.CompanionModal,
+        props: {
+          url: data?.post?.permalink,
+          onReadArticleClick: openArticle(data),
+          onActivateCompanion: activateCompanion,
+        },
+      });
+
+      trackEvent({
+        event_name: `open companion permission popup`,
+        extra: JSON.stringify({ origin: 'automatic' }),
+      });
+    },
+    [openArticle, activateCompanion, openLazyModal, trackEvent],
+  );
+
+  const articleClickHandler = useCallback(
+    async (
+      e: React.MouseEvent,
+      { post, index, row, column }: Partial<CompanionTriggerProps> = {},
+    ) => {
+      // the check whether the user is logged in is done on the GrowthBook side
+      // the feature flag is enabled only for logged-in users
+      // -- check that this is correct
+      if (!isExtension || alreadyCompleted || contentScriptGranted) {
+        await customPostClickHandler(post, index, row, column);
+      } else if (
+        // checking this also enrolls the user in the experiment
+        // so this must be the last check after all others
+        gb.isOn(Features.EngagementLoopJuly2023Companion.id)
```

Comment:
@idoshamun FYI, this method is acceptable?

Reply:
Yes

Reply:
From what I understand this will be called upon click only. According to this document https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/568492046/Companion+Install+Experiment+Design#Enrolment-Event
The enrollment event should already be dispatched on the feed visit, even if the user hasn't clicked on anything. @DragosIorgulescu is my understanding of the current implementation correct?
I'm double checking with Dave and I'll get back to you.

Reply:
Only extension feed (so excluding squad eg) version where it's applicable correct

Reply:
👍  rolled back to the initial implementation which triggers enrollment when the hook first loads

Reply:
Yep we just discussed, Dragos will propse a new flow where it's only limited to extension for enabling users

Reply:
apologies for the back and forth here, the current implementation should be correct: enrollment is triggered when landing on the feed only on the extension